File: pull.py
import zmq

context = zmq.Context()
socket = context.socket(zmq.PUSH)
socket2 = context.socket(zmq.PULL)

socket.bind("tcp://127.0.0.1:3001")
socket2.connect("tcp://127.0.0.1:3000")

while(True):
    message = 0 
    message = socket2.recv()
    if (message != 0):
        print("Received request: %s" % message)
    print("Sending Hello ")
    socket.send(b"Hello")

File: pull.js
var zmq = require("zeromq"),
  sock = zmq.socket("pull");
  sock2 = zmq.socket("push")

sock.connect("tcp://127.0.0.1:3000");
sock2.bindSync("tcp://127.0.0.1:3001");
console.log("Worker connected to port 3000");

setInterval(function() {
    console.log("sending work");
    sock2.send("some work from puller to pusher");
  }, 500);

sock.on("message", function(msg) {
  console.log("work from push: %s", msg.toString());
});